### PR TITLE
Feature 331505 Support <inheritdoc /> with no attributes

### PR DESCRIPTION
**Pull request by luyajun0205**
Feature [AB#331505](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/331505) - Support <inheritdoc /> with no attributes

Update ECMA2Yaml and Intellisense tool

### CODE REVIEW

File: ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs [hunk 1]
```diff
@@ -1172,5 +1187,236 @@ private string RemoveindexerFromPropertyName(string str)
...
+            if (!string.IsNullOrEmpty(docs?.Summary))
+            {
+                OPSLogger.LogUserWarning(LogCode.ECMA2Yaml_Inheritdoc_InvalidTags, filePath, uid);
+                return false;
+            }
+
+            return true;
+        }
+
+        private string GetInheritDocSearchId(Member m)
```
Comment (TianqiZhang) [inline ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs]: do we still need this function?

File: ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs [hunk 2]
```diff
@@ -1172,5 +1187,236 @@ private string RemoveindexerFromPropertyName(string str)
...
+            }
+            else
+            {
+                // No support other types
+            }
+
+            DoValidation(m.Docs, m.Uid, m.SourceFileLocalPath);
+        }
+
+        private bool SetInheritDoc(Docs inheritFrom, Docs inheritTo)
```
Comment (TianqiZhang) [inline ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs:1364]: can we try to clone these properties?

File: ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs (hunk 1 above)
Comment (luyajun0205) [reply ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs]: update.

File: ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs (hunk 2 above)
Comment (luyajun0205) [reply ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs:1364]: updated.

File: ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs [hunk 3]
```diff
@@ -1172,5 +1187,233 @@ private string RemoveindexerFromPropertyName(string str)
...
             return str;
         }
+
+        #region Inherit docs
+        private void BuildInheritanceForDocs(Type t)
+        {
+            var allInheritedMembersById = new Dictionary<string, List<string>>();
+            if (t.BaseTypes != null)
+            {
+                t.InheritanceChains = BuildInheritanceChain(t.Uid);
```
Comment (TianqiZhang) [inline ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs]: do we need to rebuild the inheritance chain here?

Comment (luyajun0205) [reply ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs]: updated.

File: ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs [hunk 4]
```diff
@@ -1172,5 +1187,256 @@ private string RemoveindexerFromPropertyName(string str)
...
+            bool isInherit = false;
+
+            if (!string.IsNullOrEmpty(inheritFrom?.Summary))
+            {
+                inheritTo.Summary = inheritFrom.Summary;
+                isInherit = true;
+            }
+            if (inheritFrom?.Parameters?.Count > 0)
+            {
+                inheritTo.Parameters = new Dictionary<string, string>();
```
Comment (TianqiZhang) [inline ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs]:
```suggestion
                inheritTo.Parameters = new Dictionary<string, string>(inheritFrom.Parameters);
```

Comment (luyajun0205) [reply ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs]: updated.

**Review by TianqiZhang — APPROVED** [review]

**Pull request merged**